### ISSUE TITLE
Create an enhanced Compensation simulation screen variant with htmx or similar

### ISSUE THREAD

**Issue by thusc** [smartcoop/design#154]
Once https://github.com/smartcoop/design/issues/153 is done, we can use htmx or similar to ["boost"](https://htmx.org/attributes/hx-boost/) it.

Please note this should be done as a variant of the existing screens, not as a replacement : the result will be used to show that it is possible to make such enhanced variant, not to mandate that it is the right way to use the design system.

**Comment by Wolfr** [comment]
Did any work happen towards this? Is it in another repo?

**Comment by thusc** [comment]
Emily has started working on this in the [`htmx-enhancement`](https://github.com/smartcoop/design/tree/feature/htmx-enhancement) branch.

**Comment by Wolfr** [comment]
I checked out that code, but how do I see the result? It's not clear to me since when I go to https://design.smart.coop/development/tools/simulation.html on that branch it's a “static” version (showing the results).

**Comment by thusc** [comment]
Indeed, that domain publishes only the `main` branch; we don't have a per branch subdomain yet.

**Comment by Wolfr** [comment]
I mean that I don't see it locally on that URL, sorry!

**Comment by thusc** [comment]
I think it's there: https://github.com/smartcoop/design/commit/b0a6c57257c378d817e032df1e9ce619888fab69#diff-db689e3993fe982df99815e677bfab5fc177355dc6ebcf94c51032ad5a4ec301

**Comment by emilyvandewalle** [comment]
I worked a bit more on this:

## Dependent fields
- For now, I have added the helper text for the dependent fields for the non dynamic version 
- (I also had an [alternative](https://www.figma.com/file/Zp551KO9VgmqSIouXQahlA/Smart---Screens-v1.1?node-id=1265%3A8828) to group the related fiels visually instead)
- I could maybe experiment a bit more with Alpine js to show/hide the fields based on the selection for an enhanced version?

## htmx
- local link: http://localhost:3001/development/tools/simulation-enhanced.html
- I moved the loading state to the button (will cherry pick the loader button to the `main` branch as well)
- I added a reset button to reset the form by using input `type="reset"`

## Alpine
- local link: http://localhost:3001/development/tools/simulation-enhanced-alpine.html
- Also did an experiment to load a html partial with alpine (similar to the htmx example)

## Relevant commits
- fb15bb3763858214d96377c3d9e0a4a0632acdc7 (htmx update)
- f90d968b16110b9ba9418f5ae4efb245f52b6a6d (alpine update)

**Comment by thusc** [comment]
Thank you.

> For now, I have added the helper text for the dependent fields for the non dynamic version.

Thanks, I think it's possible to make it quite good, even if a more dynamic version would be probably needed (or your alternative proposal).

- Do you think we could format the helper text as something like (so the format of those sentences are more predictable, and it is easier to scan them for the value of the field, and the name of the field):

1. Ignored when you selected `continue my Smart activity` for the `Service Smart` field.
2. Please fill this zone when you selected `interim assignment on behalf of a client` for the `Service Smart` field.

(Probably we need a better word than "zone", and something else than simply a monospace font.)

- The helper text under "Join comission" could also be formatted more similarly to the other one, e.g. in gray too. (Btw, there is a typo, it should be "commission".)
- Maybe we could have, in addition to that grey styling, a little icon or something to help our eyes catch those "optional" text more easily.

**Comment by thusc** [comment]
> (I also had an alternative to group the related fields visually instead)

I like that. This is actually a way to structure exclusive "zones" I had in mind (which I think is actually used e.g. on paper tax forms).

Do you think it's possible to find a way to make it more clear that each zone is paired with its radio button ? Maybe the radio button could be within the group border, or the text could more resemble like a title, maybe the radio button itself could be a bit more prominent (to go along the kind of title), ...

**Comment by thusc** [comment]
I think alignment could be improved a bit:
![screenshot](https://user-images.githubusercontent.com/45588452/142728868-18215948-7e60-4929-a5f4-85004fc16ac1.png)

**Comment by thusc** [comment]
The `htmx-enhancement` branch seems broken for me (when I'm using the static site): it will complain with a 404 for `/_partials/simulation-result.html` when clicking on the button. Maybe this has to do with the `_` prefixed directories ?

**Comment by emilyvandewalle** [comment]
> Do you think we could format the helper text as something like (so the format of those sentences are more predictable, and it is easier to scan them for the value of the field, and the name of the field):
> 
> 1. This field is ignored when you selected `continue my Smart activity` for the `Service Smart` field.
> 2. This zone is required when you selected `interim assignment on behalf of a client` for the `Service Smart` field.
> 
> (Probably we need a better word than "zone", and something else than simply a monospace font.)
> 

### Update 

- I had a look at your request for the helper text and tried a few things, I landed on this:
  - **Local link**: http://localhost:3000/development/tools/simulation.html
  - **Reference image:**
  
    ![image](https://user-images.githubusercontent.com/36261498/143606750-47c372be-2924-42c9-8e14-1b37e33d54d5.png)
 

> Maybe we could have, in addition to that grey styling, a little icon or something to help our eyes catch those "optional" text more easily.

I also took this into account, but I thought since the helper text is already visually heavier because of these highlighted values (and the screen in general is quite dense) it could make things more cluttered by adding more helper icons, what do you think?

**Comment by emilyvandewalle** [comment]
> > (I also had an alternative to group the related fields visually instead)
> 
> I like that. This is actually a way to structure exclusive "zones" I had in mind (which I think is actually used e.g. on paper tax forms).
> 
> Do you think it's possible to find a way to make it more clear that each zone is paired with its radio button ? Maybe the radio button could be within the group border, or the text could more resemble like a title, maybe the radio button itself could be a bit more prominent (to go along the kind of title), ...

I have processed this and tried to make the groups more clear in these designs: https://www.figma.com/file/xzzLf5aoluyutuX6Vhpt47/Smart---Library-%2B-Screens-1.2?node-id=1202%3A13539

I also disabled the fields when the option is not selected in the designs.

Let me know your feedback and if I should provide this in code

**Comment by emilyvandewalle** [comment]
I also made an experiment with Alpine to show the variant where we only show the fields when the relevant option for the related field is selected:
http://localhost:3000/development/tools/simulation-enhanced-alpine.html

**Comment by emilyvandewalle** [comment]
> The `htmx-enhancement` branch seems broken for me (when I'm using the static site, I haven't tried the development server): it will complain with a 404 for `/_partials/simulation-result.html` when clicking on the button. Maybe this has to do with the `_` prefixed directories ? (This is true for both the htmx and alpine links.)

I have removed the `_` from the folder, let me know if this fixes the issue

**Comment by emilyvandewalle** [comment]
> I think alignment could be improved a bit: ![screenshot](https://user-images.githubusercontent.com/45588452/142728868-18215948-7e60-4929-a5f4-85004fc16ac1.png)
> 

The alignment should be fixed now
